packages/flutter_tools/lib/src/android/migrations/top_level_gradle_build_file_migration.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../project.dart';
+
+const String _eagerCleanTaskDeclaration = '''
+task clean(type: Delete) {
+    delete rootProject.buildDir
+}
+''';
+
+const String _lazyCleanTaskDeclaration = '''
+tasks.register("clean", Delete) {
+    delete rootProject.buildDir
+}
+''';
+
+/// Migrate the Gradle "clean" task to use modern, lazy declaration style.
+class TopLevelGradleBuildFileMigration extends ProjectMigrator {
+  TopLevelGradleBuildFileMigration(
+    AndroidProject project,
+    super.logger,
+  ) : _topLevelGradleBuildFile = project.hostAppGradleRoot.childFile('build.gradle');
```

Comment:
@camsim99 @reidbaker @gnprice I can review the "migration" part (that LGTM) but I don't know anything about the gradle part of this.
This only changes `android/build.gradle` but not `android/app/build.gradle`, do they both need to be changed?

Reply:
Yeah, changing just the one is correct. That's the only one where the templates happen to contain the pattern that generates the Gradle warning in #122289.